spacy/tests/doc/test_span_group.py

```diff
@@ -0,0 +1,401 @@
+        concat_span_groups(span_groups)
+
+def test_span_doc_delitem(doc) :
+    span_group = doc.spans['SPANS']
+    length = len(span_group)
+    index = 5
+    span = span_group[index]
+    next_span = span_group[index + 1]
+    del span_group[index]
+    assert len(span_group) == length - 1
+    assert span_group[index] != span
+    assert span_group[index] == next_span
+
+    with pytest.raises(IndexError) :
+        del span_group[-100]
+    with pytest.raises(IndexError):
+        del span_group[100]
+
+def test_span_group_add(doc) :
+    span_group_1 = doc.spans['SPANS']
+    spans = [doc[0:5], doc[0:6]]
+    span_group_2 = SpanGroup(doc, name='MORE_SPANS', attrs={'key': 'new_value', 'new_key': 'new_value'}, spans=spans)
+
+    span_group_3_expected = span_group_1.concat(span_group_2)
+
+    span_group_3 = span_group_1 + span_group_2
+    assert len(span_group_3) ==  len(span_group_3_expected)
+    assert span_group_3.attrs == {'key': 'value', 'new_key': 'new_value'}
+    assert_span_list_equal(span_group_3, span_group_3_expected)
+
+    span_group_3 = span_group_1 + spans
+    assert len(span_group_3) == len(span_group_3_expected)
+    assert span_group_3.attrs == {'key': 'value'}
+    assert_span_list_equal(span_group_3, span_group_3_expected)
+
+
+def test_span_group_iadd(doc) :
+    span_group_1 = doc.spans['SPANS'].clone()
+    spans = [doc[0:5], doc[0:6]]
+    span_group_2 = SpanGroup(doc, name='MORE_SPANS', attrs={'key': 'new_value', 'new_key': 'new_value'}, spans=spans)
+
+    span_group_1_expected = span_group_1.concat(span_group_2)
+
+    span_group_1 +=  span_group_2
+    assert len(span_group_1) == len(span_group_1_expected)
+    assert span_group_1.attrs == {'key': 'value', 'new_key': 'new_value'}
+    assert_span_list_equal(span_group_1, span_group_1_expected)
+
+    span_group_1 = doc.spans['SPANS'].clone()
+    span_group_1 += spans
+    assert len(span_group_1) == len(span_group_1_expected)
+    assert span_group_1.attrs == {'key': 'value',}
+    assert_span_list_equal(span_group_1, span_group_1_expected)
+
+def test_span_group_extend(doc) :
+    span_group_1 = doc.spans['SPANS'].clone()
+    spans = [doc[0:5], doc[0:6]]
+    span_group_2 = SpanGroup(doc, name='MORE_SPANS', attrs={'key': 'new_value', 'new_key': 'new_value'}, spans=spans)
+
+    span_group_1_expected = span_group_1.concat(span_group_2)
+
+    span_group_1.extend(span_group_2)
+    assert len(span_group_1) ==  len(span_group_1_expected)
+    assert span_group_1.attrs == {'key': 'value', 'new_key': 'new_value'}
+    assert_span_list_equal(span_group_1, span_group_1_expected)
+
+    span_group_1 = doc.spans['SPANS']
+    span_group_1.extend(spans)
+    assert len(span_group_1) == len(span_group_1_expected)
+    assert span_group_1.attrs == {'key': 'value'}
+    assert_span_list_equal(span_group_1, span_group_1_expected)
+
+def test_span_group_dealloc(span_group) :
+    with pytest.raises(AttributeError) :
+        print(span_group.doc)
```

Comment:
Generic note: this file still requires `black` formatting (I know you know the drill by now ;-))

Reply:
I set up automatic formatting according to the instructions you gave me. All good now :)